tests/components/garmin_connect/test_config_flow.py

```diff
@@ -95,12 +101,19 @@ async def test_unknown_error(hass, mock_garmin_connect):
     assert result["errors"] == {"base": "unknown"}
 
 
-async def test_abort_if_already_setup(hass, mock_garmin_connect):
+async def test_abort_if_already_setup(hass):
     """Test abort if already setup."""
-    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_CONF, unique_id=MOCK_CONF[CONF_ID])
-    entry.add_to_hass(hass)
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=MOCK_CONF
-    )
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == "already_configured"
+    MockConfigEntry(
+        domain=DOMAIN, data=MOCK_CONF, unique_id=MOCK_CONF[CONF_ID]
+    ).add_to_hass(hass)
+    with patch(
+        "homeassistant.components.garmin_connect.config_flow.Garmin",
+    ) as garmin:
+        garmin.return_value.get_full_name.return_value = MOCK_CONF[CONF_ID]
+        yield garmin.return_value
```

Comment:
We don't need to yield here.
```suggestion
```

Reply:
Hmm run into 'TypeError: object MagicMock can't be used in 'await' expression' need to look into this...

Reply:
We may need to set `autospec` to `True` in the patch of the client class to property mock coroutine functions.